gremlin-core/pom.xml

```diff
@@ -101,6 +101,16 @@ limitations under the License.
             <artifactId>hamcrest</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```

Comment:
it looks like there is relatively minor usage of guava and lombok. could you please adjust the `StepOutputArityPredictor` to   not use these libraries so that we don't need to take the dependency?

Reply:
done